src/lib.rs

```diff
@@ -203,23 +204,125 @@ impl From<std::io::ErrorKind> for Error {
     }
 }
 
+bitflags! {
+    /// Network device flags.
+    pub struct IfFlags: u32 {
+        /// Set if the device is a loopback interface
+        const LOOPBACK = raw::PCAP_IF_LOOPBACK;
+        /// Set if the device is up
+        const UP = raw::PCAP_IF_UP;
+        /// Set if the device is running
+        const RUNNING = raw::PCAP_IF_RUNNING;
+        /// Set if the device is a wireless interface; this includes IrDA as well as radio-based
+        /// networks such as IEEE 802.15.4 and IEEE 802.11, so it doesn't just mean Wi-Fi
+        const WIRELESS = raw::PCAP_IF_WIRELESS;
+    }
+}
+
+impl From<u32> for IfFlags {
+    fn from(flags: u32) -> Self {
+        IfFlags::from_bits_truncate(flags)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Indication of whether the adapter is connected or not; for wireless interfaces, "connected"
+/// means "associated with a network".
+pub enum ConnectionStatus {
+    /// It's unknown whether the adapter is connected or not
+    Unknown,
+    /// The adapter is connected
+    Connected,
+    /// The adapter is disconnected
+    Disconnected,
+    /// The notion of "connected" and "disconnected" don't apply to this interface; for example, it
+    /// doesn't apply to a loopback device
+    NotApplicable,
+}
+
+impl From<u32> for ConnectionStatus {
+    fn from(flags: u32) -> Self {
+        match flags & raw::PCAP_IF_CONNECTION_STATUS {
+            raw::PCAP_IF_CONNECTION_STATUS_UNKNOWN => ConnectionStatus::Unknown,
+            raw::PCAP_IF_CONNECTION_STATUS_CONNECTED => ConnectionStatus::Connected,
+            raw::PCAP_IF_CONNECTION_STATUS_DISCONNECTED => ConnectionStatus::Disconnected,
+            raw::PCAP_IF_CONNECTION_STATUS_NOT_APPLICABLE => ConnectionStatus::NotApplicable,
+            // DeviceFlags::CONNECTION_STATUS should be a 2-bit mask which means that the four
+            // values should cover all the possibilities.
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct DeviceFlags {
+    pub if_flags: IfFlags,
+    pub connection_status: ConnectionStatus,
```

Comment:
`status` ?

Reply:
`status` is not clear enough and `connection_status` mirrors the original. I think this one is better left alone.